www/source/partials/docs/_internals-bootstrapping.html.md.erb

```diff
@@ -5,164 +5,339 @@ This document provides developer documentation on how the Habitat system becomes
 
-$ time record ${DB_PREFIX}base make base
+As before, to help set up environment variables for the Studio, we're going to use a [stage3 Studio](https://github.com/habitat-sh/core-plans/blob/master/bin/bootstrap/stage3-studio.sh) wrapper:
+
+```shell
+# Install only stage2 packages when setting up this Studio--that's why we're
+# calling `new` explicitly before using
+$ env HAB_STUDIO_BACKLINE_PKG=core/hab-backline/0.56.0/20180322181801 ./core-plans/bin/bootstrap/stage3-studio.sh new
+# Enter the stage3 Studio
+$ ./core-plans/bin/bootstrap/stage3-studio.sh enter
 ```
 
+Now that we're in a Studio, it's been set up with an empty artifact cache. We're going to set things up so that our install logic won't need to call out to a Builder API:
+
 ```studio
-$ build libarchive
-$ build libsodium
-$ build zeromq
-$ build ../components/depot
+# Copy all stage2 packages into the artifact cache
+$ cp ./results-stage2/*.hart /hab/cache/artifacts/
+# Install all stage2 packages directly from the artifact cache
+$ ls -1 /hab/cache/artifacts/*.hart | while read hart; do hab pkg install $hart; done
 ```
 
+Okay, time to build all the remaining plans. As before, there is a wrapper to help here:
+
 ```studio
-$ hab origin key upload -z userkey -u http://127.0.0.1:9632/v1 \
-  /hab/cache/keys/core-*.pub
-$ ls -1 launch-stage2-base-harts/*.hart \
-  | while read hart; do hab pkg up -z userkey -u http://127.0.0.1:9632/v1 $hart; done
+# (Optional) run all test suites in do_check() build phases
+$ export DO_CHECK=true
+# Build all base plans in order
+$ ./core-plans/bin/bootstrap/stage3-build-remaining-plans.sh stage3_world_order.working
 ```
 
-## Part V: Remaining packages in world
+If there are failures as you work through the remaining world, you may discover this is due to a) an older version of the software that can't build against a newer/differnt compiler, b) a failure that is currently in master and not yet caught (ex: source url now returns a 404), c) other issues which require more debugging and work. You can try to fix the package at this moment and re-run the `stage3-build-remaining-plan.sh` which will resume your failed build and continue. Alternatively, you could remove that Plan out of the rebuild set and deal with it separately. The catch here is that your failing package might become a dependency of another package later on, so you would want to remove those as well. Thankfully, there's a way you can get this list, by running another program in the `core-plans` repository. Note that you want to run this in the same common parent directory containing the `core-plans` and `habitat` git clones, and also outside of a Studio. Let's assume that the plan is for `core/doxygen`. You can run:
```

Comment:
"newer/differnt" => "newer / different"

Reply:
"not yet caught" => ??

Reply:
Re: the suggestion to do these builds in a directory that has both the habitat and core-plans repos... it'd be good to call this out earlier, and more prominently.